frontend/src/metabase/query_builder/components/view/sidebars/ModelCacheManagementSection/ModelCacheManagementSection.tsx

```diff
@@ -59,6 +61,38 @@ export function ModelCacheManagementSection({ model }: Props) {
   const isError = persistedModel?.state === "error";
   const lastRefreshTime = moment(persistedModel?.refresh_end).fromNow();
 
+  const canRefreshCache =
+    persistedModel && checkCanRefreshModelCache(persistedModel);
+
+  const refreshButton =
+    persistedModel?.state === "creating" ? (
+      <CreateOrRefreshButton
+        onClick={() => onRefresh(model.id())}
+      >{t`Create now`}</CreateOrRefreshButton>
+    ) : (
+      <CreateOrRefreshButton onClick={() => onRefresh(model.id())}>
+        <RefreshIcon name="refresh" tooltip={t`Refresh now`} />
+      </CreateOrRefreshButton>
+    );
```

Comment:
I don't think there's anything wrong with this approach but you could instead define a little component:

```
const RefreshButton = ({isModelBeingCreated}: {isModelBeingCreated: boolean}) => 
	<CreateOrRefreshButton onClick={() => onRefresh(model.id())}>
	  {isModelBeingCreated ? t`Create now` : <RefreshIcon name="refresh" tooltip={t`Refresh now`} />}
	</CreateOrRefreshButton>
```

Reply:
You could also call this outer component `CreateOrRefreshButton` and the styled component `StyledCreateOrRefreshButton`. Or for simplicity delete the styled component and use a `Button`:

```ts
import { Button } from "metabase/ui";

const CreateOrRefreshButton = ({isModelBeingCreated}: {isModelBeingCreated: boolean}) => 
	<Button onClick={() => onRefresh(model.id())} c="var(--mb-color-text-medium)">
	  {isModelBeingCreated ? t`Create now` : <RefreshIcon name="refresh" tooltip={t`Refresh now`} />}
	</Button>
```

The `Button` component has a `variant` prop you could experiment with - maybe try setting it to "subtle". I copied over the color from the styled component but you probably don't need to use that color.

Reply:
I did this and deleted all the other styled components while I was at it

https://github.com/metabase/metabase/pull/48961/commits/336b717d6825d4a5c3f575d859f4711dbc7df9fc